xarray/structure/concat.py

```diff
@@ -746,9 +746,9 @@ def get_indexes(name):
 
     # create concatenation index, needed for later reindexing
     file_start_indexes = np.append(0, np.cumsum(concat_dim_lengths))
-    concat_index = np.arange(file_start_indexes[-1])
-    concat_index_size = concat_index.size
+    concat_index_size = np.sum(concat_dim_lengths)
```

Comment:
We might squeeze a bit more, if we combine the calculation of the sum with the above np.cumsum.

Reply:
We might even think about adding the np.cumsum - trick you did further below and pre-allocate file_start_indexes. Not sure how much that gives, though.